sw/host/sphincsplus/lib.rs

```diff
@@ -2,16 +2,111 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-use sphincsplus_bindgen::SPX_BYTES;
+use sphincsplus_bindgen::crypto_sign_seed_keypair;
+use sphincsplus_bindgen::crypto_sign_keypair;
+use sphincsplus_bindgen::crypto_sign_signature;
+use sphincsplus_bindgen::crypto_sign_verify;
 
-pub fn do_something() -> u32 {
-    SPX_BYTES
+// SPHINCS+ secret key byte length. 
+pub const SPX_SECRET_KEY_BYTES : usize = sphincsplus_bindgen::CRYPTO_SECRETKEYBYTES as usize;
+pub const SPX_PUBLIC_KEY_BYTES : usize = sphincsplus_bindgen::CRYPTO_PUBLICKEYBYTES as usize;
+pub const SPX_SIGNATURE_BYTES : usize = sphincsplus_bindgen::CRYPTO_BYTES as usize;
+pub const SPX_SEED_BYTES : usize = sphincsplus_bindgen::CRYPTO_SEEDBYTES as usize;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpxPublicKey([u8; SPX_PUBLIC_KEY_BYTES]);
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpxSecretKey([u8; SPX_SECRET_KEY_BYTES]);
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpxSignature([u8; SPX_SIGNATURE_BYTES]);
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpxSeed([u8; SPX_SEED_BYTES]);
+
+// Generate a new keypair from a seed. 
+pub fn spx_keypair_from_seed(seed: &SpxSeed) -> (SpxPublicKey, SpxSecretKey) {
+    let mut pk = [0u8; SPX_PUBLIC_KEY_BYTES];
+    let mut sk = [0u8; SPX_SECRET_KEY_BYTES];
+    let err = unsafe { crypto_sign_seed_keypair(pk.as_mut_ptr(), sk.as_mut_ptr(), seed.0.as_ptr()) };
+    if err != 0 {
+        panic!("SPHINCS+ seeded key pair generation failed with error code {}.", err);
+    }
+    (SpxPublicKey(pk), SpxSecretKey(sk))
+}
```

Comment:
Nit: would be nice to make this function fallible instead of panicking

```suggestion
pub fn spx_keypair_from_seed(seed: &SpxSeed) -> anyhow::Result<(SpxPublicKey, SpxSecretKey)> {
    let mut pk = [0u8; SPX_PUBLIC_KEY_BYTES];
    let mut sk = [0u8; SPX_SECRET_KEY_BYTES];
    let err = unsafe { crypto_sign_seed_keypair(pk.as_mut_ptr(), sk.as_mut_ptr(), seed.0.as_ptr()) };
    anyhow::ensure!(err != 0, "SPHINCS+ seeded key pair generation failed with error code {err}.");
    Ok((SpxPublicKey(pk), SpxSecretKey(sk)))
}
```

Reply:
Fixed, thanks!